### PR TITLE
fix(sdk/go): ensure empty string cannot be passed as path parameters

### PR DESCRIPTION
**Pull request by GeorgeMac**
Supports https://github.com/flipt-io/flipt/pull/3716

@erka spotted that empty string can be passed to `GetNamespaces` which leads to the HTTP client making a list namespaces operation when the key is empty. This is due to the fact we strip slash on our paths.

This updates the SDK generator to support a new protobuf method option `flipt.options.flipt_client`.
This option includes the ability to define path defaults (strings only currently).
When a default is found in the HTTP client generation, it is set to the default if the path parameter is the zero value ("").

### CODE REVIEW

**Review by markphelps — APPROVED** [review]

**Comment by GeorgeMac** [comment]
I expect this will fail. We may need to merge after @erka original PR / update more code paths.
We will see when CI finishes.

**Comment by GeorgeMac** [comment]
Bare with me while I fix these tests

**Comment by GeorgeMac** [comment]
I just pushed an update to this which takes a different approach.
I will adjust the description to match.

TL;DR exploring setting path defaults as protobuf options.

File: sdk/go/http/flipt.sdk.gen.go [hunk 1]
```diff
@@ -86,6 +86,9 @@ func (x *FliptClient) BatchEvaluate(ctx context.Context, v *flipt.BatchEvaluatio
 
 func (x *FliptClient) GetNamespace(ctx context.Context, v *flipt.GetNamespaceRequest, _ ...grpc.CallOption) (*flipt.Namespace, error) {
 	var body io.Reader
+	if v.Key == "" {
+		v.Key = "default"
+	}
```
Comment (GeorgeMac) [inline sdk/go/http/flipt.sdk.gen.go:91]: See here for example output for new option.

**Review by markphelps — APPROVED** [review]
Nice
